packages/@markuplint/rule-textlint/src/index.ts

```diff
@@ -1,33 +1,98 @@
 import { Result, createRule } from '@markuplint/ml-core';
-import { TextlintKernel } from '@textlint/kernel';
+import { TextlintKernel, TextlintResult } from '@textlint/kernel';
+import { TextlintKernelOptions } from '@textlint/kernel/lib/textlint-kernel-interface';
 
-// @ts-ignore / This has not types
+// @ts-ignore / This has no types
 import TextlintPluginHTML from 'textlint-plugin-html';
 
-const kernel = new TextlintKernel();
+type TextLintEngine = import('textlint').TextLintEngine;
 
-type Option = Partial<Parameters<TextlintKernel['lintText']>[1] & {}>;
+type Option = Partial<TextlintKernelOptions> | true;
+
+const isNode = typeof process !== 'undefined' && !!process?.versions?.node;
```

Comment:
@JounQin 

It really seems to detect whether it is Node.js.
But, Will it successful to transpile through Webpack?
The plugin should ensure to run universally (isomorphically).

Reply:
I tried at https://replit.com/@JounQin/webpack-process#index.js, it is working as expected.

Reply:
@JounQin 

Not that, `isNode` will not be a problem.
Sorry, my description was bad.

`import` (or `require`) Node.js native API will be a problem through Webpack.

Reply:
OK, then we can provide different entries for Node and browser.

https://github.com/webpack/webpack/issues/8826#issuecomment-671402668

Reply:
@JounQin 

Wow, nice information.
Umm, It is maybe difficult, would you create the test that checks able to run on both Node.js and browser?

Reply:
Of course, I'll reactor it tonight and here is and example from `axios` https://github.com/axios/axios/blob/master/package.json#L69-L71, if you still want to ensure, I can create a live demo for you.

Reply:
@JounQin 

Ok, It's up to you, thanks.

Reply:
Done! See https://github.com/markuplint/markuplint/pull/163/files#diff-1d0dc22e273f9130ccfc415beec3c85eab51ac8e36d5355a9de1beaf149b8efeR86-R89.